### ISSUE TITLE
How to get the imputed data?

### ISSUE THREAD

**Issue by TaichiLi** [ProcessMonitoringStellenboschUniversity/IFAC-VAE-Imputation#1]
I run this code with my data, but I can't get the data which be imputed, I have only some picture. I don't know how to get the data.Could you please tell the way to get the data?

**Comment by skroon** [comment]
Can you give us the exact commands you issued? Perhaps @JTMcCoy will be able to help you out then.

**Comment by TaichiLi** [comment]
emm.. The code is excellent. My question is:
The original data and the corrupted data have been standardized. Then how do I get the data with original form ?@JTMcCoy

**Comment by JTMcCoy** [comment]
I used sklearn's StandardScaler to standardise the data (lines 91 and 93 of ```main.py```) and then use the ```inverse_transform``` method of the StandardScaler to transform the data back to its original scale (line 181 of ```main.py```). Hope that helps.

**Comment by TaichiLi** [comment]
Thanks. Could you please tell me the function of `ImputerIter`?

**Comment by JTMcCoy** [comment]
```ImputeIter``` is a parameter to control how many iterations of imputation are used by the VAE; I found for my data that imputed values were converging within 25 iterations, if the network had converged, and so I set ```ImputeIter = 25```. The figure plotted from line 155 in ```main.py``` shows the convergence of the imputations with imputation iterations, so you can use this to check visually whether the value you choose is appropriate for your dataset.

I've just updated the paper associated with this repo (you can access it free of charge here: https://doi.org/10.1016/j.ifacol.2018.09.406). Hopefully this will help to describe some more of the technical detail.

**Comment by TaichiLi** [comment]
I have another question. If I want to get better result, should I do some work to process my data? And when I train model, I found a strange phenomenon.
![Figure_1](https://user-images.githubusercontent.com/47553337/54931344-c6efe080-4f53-11e9-9ad2-7710682e16e6.png)
The ELBO drops suddenly, what are the possible reasons? Thank you for your patient answers

**Comment by JTMcCoy** [comment]
It's always a good idea to do some data processing, but I'm not sure exactly what you mean by your question. For inputs to a neural net, I'd definitely suggest standardising or normalising in some way! I found that if I didn't standardise inputs to this VAE, it very quickly hit ```NaN``` and ```Inf``` during training.

That is an interesting drop in the ELBO, I presume it's due to the stochastic nature of training (mini-batches etc). I found on some datasets that the ELBO went through a few plateaus, so there are some interesting trends at times, and the convergence of this sort of network definitely needs more investigation - I hope you'll find something as you continue to work on it!
